### ISSUE TITLE
Example on using MLAS directly

### ISSUE THREAD

**Issue by mmeterel** [ymwangg/mlas#2]
@ymwangg 
Hi,
I was trying to use MLAS directly without onnxruntime and found your repo. Do you have any example where I can use SGEMM (for example) directly using MlasGEMM API?
Thanks

**Comment by ymwangg** [comment]
Yes, this may be useful. https://github.com/ymwangg/mlas-old/blob/main/mlas_sgemm.cc

**Comment by mmeterel** [comment]
@ymwangg Thanks alot. It is very useful. I have written a similar code, but having hard time to link it. I see an error like this:

sgemm.cpp:(.text._Z13MlasGemmBatch15CBLAS_TRANSPOSES_mmmPK22MLAS_SGEMM_DATA_PARAMSmPN11onnxruntime11concurrency10ThreadPoolE+0xdf): undefined reference to `onnxruntime::concurrency::ThreadPool::DegreeOfParallelism(onnxruntime::concurrency::ThreadPool const*)'

I am building onnxruntime locally using the command below:
./build.sh --config Release --parallel --build_shared_lib

And building/linking my code as follows:

g++ -std=c++11  -c test.cpp -I<path>/onnxruntime/onnxruntime/core/mlas/lib
g++ -std=c++11  -o test test.o -I<path>/onnxruntime/onnxruntime/core/mlas/lib -L<path>/onnxruntime/build/Linux/Release/ -lonnxruntime_mlas -lonnxruntime
/bin/ld: <path>/onnxruntime/build/Linux/Release//libonnxruntime_mlas.a(sgemm.cpp.o): in function `MlasGemmBatch(CBLAS_TRANSPOSE, CBLAS_TRANSPOSE, unsigned long, unsigned long, unsigned long, MLAS_SGEMM_DATA_PARAMS const*, unsigned long, onnxruntime::concurrency::ThreadPool*)':

This is the first time I am working on onnxruntime, so I am not quite familiar with repo. Do you have any advice on the error above?

If the error is not familiar, I can also appreciate if you can share how you are building/linking the example you shared. 
Thanks alot.

**Comment by mmeterel** [comment]
Attached is the sample code I am trying to build. Basically, I am trying to use MlasGemm API directly. Not sure if this is possible at all. 

[test.txt](https://github.com/ymwangg/mlas/files/11826881/test.txt)

**Comment by ymwangg** [comment]
To compile my example code, you can run:
```
git clone https://github.com/ymwangg/mlas-old.git
cd mlas-old
cmake .
make -j8
./test 10 10 10
```

`undefined reference` typically means you need to manually add the corresponding source files to your target similar to here https://github.com/ymwangg/mlas/blob/2fa74e6193350f5fa146fa100a6f7eb162ed69a1/CMakeLists.txt#L28-L49
